packages/eas-cli/src/build/android/applicationId.ts

```diff
@@ -0,0 +1,158 @@
+import { ExpoConfig, getConfigFilePaths } from '@expo/config';
+import { AndroidConfig } from '@expo/config-plugins';
+import assert from 'assert';
+import chalk from 'chalk';
+import fs from 'fs-extra';
+import path from 'path';
+
+import log from '../../log';
+import {
+  getAndroidApplicationIdAsync,
+  getProjectConfigDescription,
+} from '../../project/projectUtils';
+import { promptAsync } from '../../prompts';
+import { gitAddAsync } from '../../utils/git';
+
+enum ApplicationIdSource {
+  AndroidProject,
+  AppJson,
+}
+
+export async function configureApplicationIdAsync(
+  projectDir: string,
+  exp: ExpoConfig,
+  allowExperimental: boolean
+): Promise<void> {
+  const configDescription = getProjectConfigDescription(projectDir);
+  const applicationIdFromConfig = AndroidConfig.Package.getPackage(exp);
+  const applicationIdFromAndroidProject = await getAndroidApplicationIdAsync(projectDir);
+
+  if (applicationIdFromAndroidProject && applicationIdFromConfig) {
+    if (applicationIdFromConfig !== applicationIdFromAndroidProject) {
+      log.newLine();
+      log.warn(
+        `We detected that your Android project is configured with a different application id than the one defined in ${configDescription}.`
+      );
+      if (!(await hasApplicationIdInStaticConfigAsync(projectDir, exp))) {
+        log(`If you choose the one defined in ${configDescription} we'll automatically configure your Android project with it.
+However, if you choose the one defined in the Android project you'll have to update ${configDescription} on your own.
+Otherwise, you'll see this prompt again in the future.`);
+      }
+      log.newLine();
+      const { applicationIdSource } = await promptAsync({
+        type: 'select',
+        name: 'applicationIdSource',
+        message: 'Which application id should we use?',
+        choices: [
+          {
+            title: `${chalk.bold(
+              applicationIdFromAndroidProject
+            )} - In build.gradle in Android project`,
+            value: ApplicationIdSource.AndroidProject,
+          },
+          {
+            title: `${chalk.bold(applicationIdFromConfig)} - In your ${configDescription}`,
+            value: ApplicationIdSource.AppJson,
+          },
+        ],
+      });
+      switch (applicationIdSource) {
+        case ApplicationIdSource.AppJson: {
+          await updateApplicationIdInBuildGradleAsync(projectDir, exp);
+          break;
+        }
+        case ApplicationIdSource.AndroidProject: {
+          await updateAppJsonConfigAsync(projectDir, exp, applicationIdFromAndroidProject);
```

Comment:
Maybe we should check if the user has an Expo config, and if so, quit the process and tell the user that the config doesn't match the native project, that they should re-run `expo eject` to sync automatically, or manually sync the config if eject doesn't work for some reason. This leans more towards encouraging the user to control their project via the app config. Also allows us to keep complex syncing logic in the expo/config ecosystem.

Reply:
That sth similar i initially intended, there was log above but I forgot to add the same check inside switch

`updateAppJsonConfigAsync` should not be called if `hasApplicationIdInStaticConfigAsync` returns false